### PR TITLE
1. overflow science when finishing and starting research. 2. in the cityscreen unit should be not visible. 3. modifier the policy "Commerce Complete" according to original game. 4. the action of "Great Merchant" now is the same as original game. 5. translation update.

### PR DESCRIPTION
**Pull request by lishaoxia1985**

### CODE REVIEW

File: core/src/com/unciv/logic/civilization/TechManager.kt [hunk 1]
```diff
@@ -79,8 +81,7 @@ class TechManager {
     fun remainingScienceToTech(techName: String) = costOfTech(techName) - researchOfTech(techName)
 
     fun turnsToTech(techName: String): Int {
-        return ceil( remainingScienceToTech(techName).toDouble()
-                / civInfo.statsForNextTurn.science).toInt()
+        return max(1, ceil( remainingScienceToTech(techName).toDouble() / civInfo.statsForNextTurn.science).toInt())
```
Comment (lishaoxia1985) [inline core/src/com/unciv/logic/civilization/TechManager.kt]: sometimes the turn will become 0.

File: core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt [hunk 2]
```diff
@@ -232,12 +232,18 @@ class UnitActions {
...
                 // http://civilization.wikia.com/wiki/Great_Merchant_(Civ5)
-                val goldGained = 350 + 50 * unit.civInfo.getEra().ordinal
-                unit.civInfo.gold += goldGained
-                unit.civInfo.addNotification("Your trade mission has earned you [$goldGained] gold!",null, Color.GOLD)
+                var goldGained = (350 + 50 * unit.civInfo.getEra().ordinal) * unit.civInfo.gameInfo.gameParameters.gameSpeed.getModifier()
+                if (unit.civInfo.policies.isAdopted("Commerce Complete"))
+                    goldGained *= 2
+                unit.civInfo.gold += goldGained.toInt()
+                tile.owningCity!!.civInfo.getDiplomacyManager(unit.civInfo).influence += 30
+                unit.civInfo.addNotification("You get [${goldGained.toInt()}] gold and 30 Influence with [${tile.owningCity!!.civInfo}] from trade mission!",null, Color.GOLD)
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt]: "Your trade mission has earned you [goldAmount] gold and [influenceAmount] influence!"
Nicer English

**Comment by yairm210** [comment]
It looks pretty OK, but I have to say that if you split each change into a separate branch it would be much easier to go over and accept the parts that are OK and put off only the parts that aren't yet OK

File: core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt (hunk 2 above)
Comment (lishaoxia1985) [reply core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt]: it losts the key word "otherciv".😊

Comment (yairm210) [reply core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt]: "Your trade mission to [civName] has earned you [goldAmount] gold and [influenceAmount] influence!"

Comment (lishaoxia1985) [reply core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt]: ok, before you merge other pull requests, make sure the game can run well, i fix your newest merge which  can not run.

**Pull request merged**